File: Android/app/src/main/java/com/decisionkitchen/decisionkitchen/Group.kt
package com.decisionkitchen.decisionkitchen

/**
 * Created by kevin on 2017-07-29.
 */

data public class Category(val name: String? = null, val type: String? = null, val value: Any? = null)

data public class RestaurantOrder(val id: String? = null, val service: String? = null)

data public class Restaurant(val address: String? = null, val name: String? = null, val id: String? = null, val order: RestaurantOrder? = null)

data public class GameMeta(val end: String? = null, val start: String? = null)

data public class GameResult(val restaurant_id: String? = null)

data public class Game(
    val categories: ArrayList<Category>? = null,
    val meta: GameMeta? = null,
    val rating: HashMap<String, Integer>? = null,
    val responses: HashMap<String, ArrayList<Integer>>? = null,
    val result: GameResult? = null)

data public class Group(
    val password: String? = null,
    val members: ArrayList<String>? = null,
    val name: String? = null,
    val restaurants: HashMap<String, Restaurant>? = null,
    val id: String? = null,
    val games: ArrayList<Game>? = null
)

File: iOS/DecisionKitchen/Group.swift
//
//  Group.swift
//  DecisionKitchen
//
//  Created by Andy Liang on 2017-07-29.
//  Copyright © 2017 Andy Liang. All rights reserved.
//

import UIKit

typealias UserID = String

class Group: NSObject, Codable {
    
    var name: String
    
    var password: String
    
    var members: [UserID]
    
    var restaurants: [String: Restaurant]?
    
    var games: [Game]?
    
    var id: String
    
    init(name: String, password: String, members: [UserID], restaurants: [String: Restaurant]?, games: [Game]?, id: String) {
        self.name = name
        self.password = password
        self.members = members
        self.restaurants = restaurants
        self.games = games
        self.id = id
        super.init()
    }
    
}
